Automattic/woocommerce-payments
# Add support for platform checkout server side payment processing

## Changes proposed in this Pull Request

Supersedes #3642 





This PR adds the necessary functionality for us to support platform payment processing without introducing all the new filters we needed for the first approach we tried in #3642 .

- Add a new filter, `wc_payments_account_id_for_intent_confirmation`, to allow the platform to modify the account used to confirm a payment intent.
    - **Note:** The account ID must match the provided payment intent, otherwise the intent confirmation will fail.
- Add new JS prop to the frontend settings, `accountIdForIntentConfirmation`, which contains the result of running the new filter on an empty string.
- Add the ability to process a payment when the checkout request provides a payment intent.
    - This is so the platform can process the payment and then update the order on the merchant store appropriately.






## Testing instructions





1. Make sure payments work as they normally do in a normal WCPay store. Maybe try some of the [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows)?
2. Follow the testing instructions in the corresponding platform checkout PR (should be mentioned below via GitHub mention).

I have no automated tests for this because I'm not so sure how to actually cover this with automated tests.

-------------------


**Post merge**
